crates/host/src/policy.rs

```diff
@@ -260,18 +256,97 @@ impl Manager {
         Ok(manager)
     }
 
-    /// Constructs a
     #[instrument(level = "trace", skip_all)]
-    pub async fn evaluate_action(
+    /// Use the policy manager to evaluate whether a component may be started
+    pub async fn evaluate_start_component(
+        &self,
+        component_id: impl AsRef<str>,
+        image_ref: impl AsRef<str>,
+        max_instances: u32,
+        annotations: &BTreeMap<String, String>,
+        claims: &Option<&jwt::Claims<jwt::Actor>>,
+    ) -> anyhow::Result<Response> {
+        let request = ComponentInformation {
+            component_id: component_id.as_ref().to_string(),
+            image_ref: image_ref.as_ref().to_string(),
+            max_instances,
+            annotations: annotations.clone(),
+            claims: claims.map(PolicyClaims::from),
+        };
+        self.evaluate_action(RequestBody::StartComponent(request))
+            .await
+    }
+
+    /// Use the policy manager to evaluate whether a provider may be started
+    #[instrument(level = "trace", skip_all)]
+    pub async fn evaluate_start_provider(
+        &self,
+        provider_id: impl AsRef<str>,
+        provider_ref: impl AsRef<str>,
+        annotations: &BTreeMap<String, String>,
+        claims: &Option<&jwt::Claims<jwt::CapabilityProvider>>,
+    ) -> anyhow::Result<Response> {
+        let request = ProviderInformation {
+            provider_id: provider_id.as_ref().to_string(),
+            image_ref: provider_ref.as_ref().to_string(),
+            annotations: annotations.clone(),
+            claims: claims.map(PolicyClaims::from),
+        };
+        self.evaluate_action(RequestBody::StartProvider(request))
+            .await
+    }
+
+    /// Use the policy manager to evaluate whether a component may be invoked
+    #[instrument(level = "trace", skip_all)]
+    pub async fn evaluate_perform_invocation(
         &self,
-        source: Option<RequestSource>,
-        target: RequestTarget,
-        action: Action,
+        component_id: impl AsRef<str>,
+        image_ref: impl AsRef<str>,
+        annotations: &BTreeMap<String, String>,
+        claims: &Option<&jwt::Claims<jwt::Actor>>,
```

Comment:
This (and all the other claims params in these functions) should just be `Option<&jwt::Claims<jwt::Actor>>`. You can take a borrowed option and use `as_ref()` or `as_deref` to get a borrowed inner value